lib/saluki-config/src/lib.rs

```diff
@@ -302,10 +316,15 @@ impl ConfigurationLoader {
 
     /// Enables dynamic configuration.
     ///
-    /// This will enable the dynamic configuration feature, which allows the configuration to be sourced from the core agent.
-    pub fn with_dynamic_configuration(mut self) -> Self {
+    /// This will enable the dynamic configuration feature, which allows the configuration to be sourced from a dynamic provider. The dynamic provider is updated at runtime, and `figment` will re-read from it when configuration values are requested.
+    pub fn with_dynamic_configuration(mut self) -> Result<Self, ConfigurationError> {
         self.dynamic_configuration_enabled = true;
```

Comment:
Wonder if we should just drop this field entirely since we can determine it by whether or not `dynamic_values` is `None`/`Some`?